#!/usr/bin/env python

from twython import Twython, TwythonError
from redis import Redis
import os


redis_url = os.getenv('REDISTOGO_URL', 'redis://localhost:6379')
r = Redis.from_url(redis_url)

APP_KEY = "kHLxa38Nt8sT8P9lz4ymRcgHT"
APP_SECRET = "MLZzxOoFCBCtOVCkN1yssGrEMGiR7U4BW4oGoxRphvv0zoUlWp"

token = r.get("twitter:token")
secret = r.get("twitter:secret")
twitter = Twython(APP_KEY, APP_SECRET, token, secret)

#follow by keywords 
with open("keywords.txt") as f:
  for line in f:
    keyword = line.strip()
    for i in range(10):
        people = twitter.search_users(q=keyword, page=i)
        for p in people:
            try:
              twitter.create_friendship(user_id=p['id'])
              r.hset(p['id'], 'follows', False)
              r.hset(p['id'], 'keyword', keyword)
            except TwythonError as e:
              pass


